code/src/reducers/todo.js

```diff
@@ -0,0 +1,47 @@
+import { createSlice } from '@reduxjs/toolkit'
+import uniqid from 'uniqid'
+
+const todo = createSlice({
+  name: 'todos',
+  initialState: {
+    items: [],
+  },
+  reducers: {
+    addTodo: (store, action) => {
+      const newTodo = {
+        id: uniqid(),
+        text: action.payload,
+        isComplete: false,
+      }
+
+      //Immutability approach
+      store.items = [...store.items, newTodo]
+    },
+    toggleTodo: (store, action) => {
+      //Immutability approach
+      const updatedItems = store.items.map((item) => {
+        if (item.id === action.payload) {
+          const updatedTodo = {
+            ...item,
+            isComplete: !item.isComplete,
+          }
+          return updatedTodo
+        } else {
+          return item
+        }
+      })
+
+      store.items = updatedItems
+    },
+    deleteTodo: (store, action) => {
+      //Immutability approach
```

Comment:
Like that you keep this comment to clarify which approach you went for. Both for others to see and to go back to and maybe copy in comming projects :)